_source/_posts/2021-11-08-k8s-api-server-oidc.md

```diff
@@ -0,0 +1,565 @@
+
+Copy the output values as we will need them for the next steps.
+
+```bash
+okta_group.k8s_admin: Creating...
+okta_group.k8s_restricted_users: Creating...
+okta_auth_server.oidc_auth_server: Creating...
+okta_app_oauth.k8s_oidc: Creating...
+okta_group.k8s_admin: Creation complete after 0s [id=00g2b0rmupxo1C1HW5d7]
+okta_group.k8s_restricted_users: Creation complete after 0s [id=00g2b0q4zejpq6hbi5d7]
+okta_group_memberships.restricted_user: Creating...
+okta_group_memberships.admin_user: Creating...
+okta_auth_server.oidc_auth_server: Creation complete after 1s [id=aus2b0ql0ihgilIh95d7]
+okta_auth_server_claim.auth_claim: Creating...
+okta_group_memberships.admin_user: Creation complete after 1s [id=00g2b0rmupxo1C1HW5d7]
+okta_group_memberships.restricted_user: Creation complete after 1s [id=00g2b0q4zejpq6hbi5d7]
+okta_app_oauth.k8s_oidc: Creation complete after 1s [id=0oa2b0qc0x38Xjxbk5d7]
+okta_app_group_assignments.k8s_oidc_group: Creating...
+okta_auth_server_policy.auth_policy: Creating...
+okta_auth_server_claim.auth_claim: Creation complete after 0s [id=ocl2b0rc6ejmIO4KR5d7]
+okta_app_group_assignments.k8s_oidc_group: Creation complete after 1s [id=0oa2b0qc0x38Xjxbk5d7]
+okta_auth_server_policy.auth_policy: Creation complete after 1s [id=00p2b0qjjxsSz0Mya5d7]
+okta_auth_server_policy_rule.auth_policy_rule: Creating...
+okta_auth_server_policy_rule.auth_policy_rule: Creation complete after 1s [id=0pr2b0rgnfxjyu6cy5d7]
+
+Apply complete! Resources: 10 added, 0 changed, 0 destroyed.
+
+Outputs:
+
+k8s_oidc_client_id = "0oa2b0qc0x38Xjxbk5d7"
+k8s_oidc_issuer_url = "https://dev-xxxxxx.okta.com/oauth2/aus2b0ql0ihgilIh95d7"
+```
+
+You can run `terraform destroy` to revert the changes if required.
+
+> **TIP**: You can use the `terraform import <resource_name.id>` command to import data and configuration from your Okta instance. Refer [these docs](https://registry.terraform.io/providers/okta/okta/latest/docs) for more information.
+
+## Preparing the cluster
+
+Now we need to prepare the cluster to work with OIDC. For this, we need to update the API server flags below:
+
+- `oidc-issuer-url`: This will be your issuer URL from the Okta authorization server
+- `oidc-client-id`: This is the client ID from the Okta OIDC application
+- `oidc-username-claim`: This is the claim that will be used to identify the user. In this case, it is `email`.
+- `oidc-groups-claim`: This is the claim that will be used to identify the groups. In this case, it is `groups`.
+- `oidc-ca-file`: This is the CA file that is used to validate the OIDC server. If you are using Okta as an OIDC provider, then you don't need to set this.
+
+**NOTE**: If you encounter an issue where OIDC groups/users collide with Kubernetes groups/users, you can set `oidc-groups-prefix` and `oidc-username-prefix` flags as well.
+
+Flags can be set when creating the cluster or by patching the API server via SSH.
```

Comment:
```suggestion
Flags can be set when creating the cluster or by patching the API server via SSH, as described below.
```
i got super distracted by "what does patching it even mean" and went to look it up and should have just kept reading :)